src/syntax/rules.rkt

```diff
@@ -15,9 +17,27 @@
 
 (define *rulesets* (make-parameter '()))
 
+(define (rule-ops-supported? rule)
+  (define (ops-in-expr expr)
+    (cond
+      [(list? expr) (if (member (car expr) (*loaded-ops*))
+                        (for/and ([subexpr (cdr expr)])
+                          (ops-in-expr subexpr))
+                        #f)]
+      [else #t]))
+  (and (ops-in-expr (rule-input rule)) (ops-in-expr (rule-output rule))))
```

Comment:
It doesn't make any difference right now, but I think we should leave in place rules that have "invalid" operators only as their inputs. This would then generalize to future situations like saying, "rewrite this so as not to use X".